gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala

```diff
@@ -489,6 +489,13 @@ object ExpressionConverter extends SQLConfHelper with Logging {
           rightChild,
           resultType,
           b)
+      case n: NaNvl =>
+        IfTransformer(
+          replaceWithExpressionTransformerInternal(IsNaN(n.left), attributeSeq, expressionsMap),
+          replaceWithExpressionTransformerInternal(n.right, attributeSeq, expressionsMap),
+          replaceWithExpressionTransformerInternal(n.left, attributeSeq, expressionsMap),
+          n
+        )
```

Comment:
cc @zzcclp

Reply:
I don't know, but let me to limit this replacement for velox only now, thanks.